Darwin fixes

toonn opened the pull request:
Hi, I'm working on fixing the FFmpeg 7 build in Nixpkgs on Darwin/macOS and I figured some of the changes may be welcome upstream. This is a parallel PR to mpeg5/xevd#63.

[Using `CMAKE_INSTALL_FULL_*` paths](https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html) will avoid a problem due to us setting `CMAKE_INSTALL_*` to absolute paths, which causes the manual prefixing currently done to create non-existing paths.

The change in xevem_util.c might be a bug or I might be wrong and it should indeed iterate in steps of size 2. (Similar to the xevdm_util.c change in the other PR.)

toonn commented:
[The unconditional `float32x4_t` cast on line 6006](https://github.com/mpeg5/xeve/blob/master/src_base/neon/sse2neon.h#L6006) is incorrect on `__aarch64__` because [`__m128d` is defined to be `float64x2_t` there](https://github.com/mpeg5/xeve/blob/master/src_base/neon/sse2neon.h#L215-L219).

[95b8903](https://github.com/mpeg5/xeve/pull/122/commits/95b8903b2e3ea1bb8e9d53f1745985dd912a0656) fixes this by leaning on the typedef'd `__m128d` for the cast. Alternatively we could add a branch on `__aarch64__`.

dariusz-f commented on CMakeLists.txt:
```diff
@@ -141,7 +141,7 @@ message ("linker Flags: " ${CMAKE_EXE_LINKER_FLAGS})
 ########################################
 
-set(CMAKE_C_STANDARD 99)
+set(CMAKE_C_STANDARD 11)
```
We need to keep C99 standard

emilazy replied:
The codebase already uses C11 features, so either the compiler needs to be set to use C11 or the code using C11 features has to be changed. Are there any plans to do the latter or do you want an external PR to address this?

dariusz-f replied:
#124

toonn replied:
I split this part out into #125.

toonn commented on src_base/neon/sse2neon.h:
```diff
@@ -6003,7 +6003,7 @@ FORCE_INLINE void _mm_storeu_si32(void *p, __m128i a)
 {
 #if __has_builtin(__builtin_nontemporal_store)
-    __builtin_nontemporal_store(a, (float32x4_t *) p);
+    __builtin_nontemporal_store(a, (__m128d *) p);
```
OK, I'll drop this change, we'll carry it in Nixpkgs for the time being since it's required for aarch64 support.

toonn commented on src_base/xeve_enc.c:
```diff
@@ -2139,7 +2139,6 @@ static void parse_chroma_qp_mapping_table(XEVE_CHROMA_TABLE* chroma_qp_table, XE
                     break;
                 chroma_qp_table->delta_qp_in_val_minus1[0][j++] = atoi(val);
             } while (1);
-            if (chroma_qp_table->num_points_in_qp_table_minus1[0] + 1 == j);
```
We'll similarly carry this change in Nixpkgs since disabling a warning is more or less equivalent to removing effectless code.

toonn commented:
I've dropped the requested changes and split a number of things into their own PRs. All this PR now does is make passing of the `-static` flag conditional, static linking is strongly discouraged on Darwin, and add a condition for the includes of time-related headers in `xeve_app_util.h`.

The latter I've done by appending `|| __APPLE__`. [This page](https://sourceforge.net/p/predef/wiki/OperatingSystems/) lists it as `__APPLE__ && __MACH__` but that's only if these headers shouldn't be included on, for example, iOS. I can change this if desired.
